sklearn/feature_selection/_rfe.py

```diff
@@ -721,7 +721,7 @@ def fit(self, X, y, groups=None):
             func = delayed(_rfe_single_fit)
 
         scores = parallel(
-            func(rfe, self.estimator, X, y, train, test, scorer)
+            func(rfe, clone(self.estimator), X, y, train, test, scorer)
```

Comment:
Can we construct a non-regression test for this?

Reply:
Hi @thomasjpfan, I tried to reproduce the error in the original issue but wasn't able to do so :(. Could you help take a look if you got time, thank you!

Reply:
I am not sure if this is where the original error comes from.  `func = _rfe_single_fit` does not actually mutate `estimator`:

https://github.com/scikit-learn/scikit-learn/blob/2f787f4c99401120184649cdc5c3e098f9823746/sklearn/feature_selection/_rfe.py#L31

`_rfe_single_fit` only uses `estimator` for `_safe_split` and the lambda function has it's own locally scoped `estimator.

https://github.com/scikit-learn/scikit-learn/blob/2f787f4c99401120184649cdc5c3e098f9823746/sklearn/feature_selection/_rfe.py#L40-L42

Reply:
Shall I close this PR for now ? Since it seems not related to the original issue

Reply:
Yea, I am okay with closing.